### PR TITLE
fix(Unstable_Button): override previous theme styles

### PR DESCRIPTION
**Pull request by shadoath**
- Specifically targeting when the theme Provider of spark v1 has styling overrides.

### CODE REVIEW

File: libs/spark/CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,9 @@
 
 ## [vNext](https://github.com/prenda-school/prenda-spark/compare/v1.0.0-alpha.5...vNext) (yyyy-mm-dd)
 
+- **Unstable_Button**
+  - Update styling on Unstable button to match Figma mockups. Required overriding the SparkThemeProvider styles.
+
```
Comment (WilliamKelley) [inline libs/spark/CHANGELOG.md:7]:
```suggestion
### Fixes

- **Unstable_Button**
  - Override conflicting styles from previous `theme` (e.g. when rendered under `SparkThemeProvider`) for pseudo states like hover, disabled, etc.
```

File: libs/spark/src/Unstable_Button/Unstable_Button.tsx [hunk 2]
```diff
@@ -79,13 +79,20 @@ const buttonFontVariantLarge = buildVariant(
...
       borderRadius: 4,
       borderStyle: 'solid',
       borderWidth: 1,
       '&.Mui-focusVisible, &:focus-visible': {
         boxShadow: `0px 0px 2px 4px ${theme.unstable_palette.teal[300]}`,
       },
+      // Double specificity to override V1 theme style
+      '&&:disabled': {
+        opacity: 'unset',
+      },
```
Comment (WilliamKelley) [inline libs/spark/src/Unstable_Button/Unstable_Button.tsx:95]:
```suggestion
      // override V1 theme style
      '&&:disabled': {
        opacity: 'unset',
      },
```
It's only to override the style. Not an existing style that needs double specificity.

File: libs/spark/src/Unstable_Button/Unstable_Button.tsx [hunk 3]
```diff
@@ -102,11 +109,13 @@ const useStyles = makeStyles<Unstable_ButtonClassKey>(
         },
       }),
       ...(props.variant === 'stroked' && {
-        borderColor: `${theme.unstable_palette.neutral[90]}`,
-        backgroundColor: 'transparent', // theme.palette.common.white,
-        // '&:hover': {
-        //   backgroundColor: theme.unstable_palette.neutral[70],
-        // },
+        '&&': {
```
Comment (WilliamKelley) [inline libs/spark/src/Unstable_Button/Unstable_Button.tsx:112]:
```suggestion
        // double specificity to override v1 theme style
        '&&': {
```

File: libs/spark/src/Unstable_Button/Unstable_Button.tsx [hunk 4]
```diff
@@ -195,6 +204,9 @@ const useStyles = makeStyles<Unstable_ButtonClassKey>(
         ...(props.disabled && {
           color: theme.unstable_palette.neutral[100],
         }),
+        '&:hover > & ': {
+          color: theme.palette.common.white,
+        },
```
Comment (WilliamKelley) [inline libs/spark/src/Unstable_Button/Unstable_Button.tsx:209]:
```suggestion
```
Pretty sure this was unnecessary.

**Review by WilliamKelley — CHANGES_REQUESTED** [review]
Title changed to comply to conventional commits.

File: libs/spark/src/Unstable_Button/Unstable_Button.tsx (hunk 4 above)
Comment (shadoath) [reply libs/spark/src/Unstable_Button/Unstable_Button.tsx:209]: Correct...whoops

**Comment by shadoath** [comment]
@WilliamKelley, I had forgot about that. It is just so nice to hit merge commit. 
I could turn the PR to draft and do it though then bring it back out to trigger only one chromatic merge right?

**Comment by shadoath** [comment]
Or use the 'batch' ill use that more.

**Comment by WilliamKelley** [comment]
Oh, it looks like even successful runs have that error message, but it doesnt exit the process with an error code and fail the job, it exits fine.

So we should follow with the workaround described in the issue

**Comment by shadoath** [comment]
So we need to do the following: 
```
variables:
  GIT_CHECKOUT: 'false'

before_script:
  - git checkout "$CI_COMMIT_REF_NAME"
  ```

**Comment by WilliamKelley** [comment]
> So we need to do the following?
> 
> ```
> variables:
>   GIT_CHECKOUT: 'false'
> 
> before_script:
>   - git checkout "$CI_COMMIT_REF_NAME"
> ```

I believe those are GitLab specific, so it would be the equivalent GitHub Actions env var? I'm not sure. 

Regardless though, it seems like it succeeded once you added an empty line between a list item and a heading! Which makes sense because the absence of that would cause format:check to fail. So it just has a poor output.

**Review by WilliamKelley — APPROVED** [review]

**Comment by WilliamKelley** [comment]
@slingner review bump. we can get this released today

**Review by slingner — APPROVED** [review]
🚀

**Pull request merged**